tools/kagami/src/main.rs

```diff
@@ -363,43 +363,120 @@ mod genesis {
+    pub enum Mode {
+        Client(client::Args),
+        Peer(peer::Args),
+    }
 
     impl<T: Write> RunArgs<T> for Args {
         fn run(self, writer: &mut BufWriter<T>) -> Outcome {
-            let config = ConfigurationProxy {
-                torii_api_url: Some(SmallStr::from_str(DEFAULT_API_URL)),
-                torii_telemetry_url: Some(SmallStr::from_str(DEFAULT_TORII_TELEMETRY_URL)),
-                account_id: Some("alice@wonderland".parse()?),
-                basic_auth: Some(Some(BasicAuth {
-                    web_login: WebLogin::new("mad_hatter")?,
-                    password: SmallStr::from_str("ilovetea"),
-                })),
-                public_key: Some(PublicKey::from_str(
-                    "ed01207233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0",
-                )?),
-                private_key: Some(PrivateKey::from_hex(
-                    Algorithm::Ed25519,
-                    "9ac47abf59b356e0bd7dcbbbb4dec080e302156a48ca907e47cb6aea1d32719e7233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0"
-                )?),
-                ..ConfigurationProxy::default()
+            match self.mode {
+                Mode::Client(args) => args.run(writer),
+                Mode::Peer(args) => args.run(writer),
+            }
+        }
+    }
+
+    mod client {
+        use iroha_config::{
+            client::{BasicAuth, ConfigurationProxy, WebLogin},
+            torii::{uri::DEFAULT_API_URL, DEFAULT_TORII_TELEMETRY_URL},
+        };
+
+        use super::*;
+
+        #[derive(StructOpt, Debug, Clone, Copy)]
+        pub struct Args;
+
+        impl<T: Write> RunArgs<T> for Args {
+            fn run(self, writer: &mut BufWriter<T>) -> Outcome {
+                let config = ConfigurationProxy {
+                    torii_api_url: Some(SmallStr::from_str(DEFAULT_API_URL)),
+                    torii_telemetry_url: Some(SmallStr::from_str(DEFAULT_TORII_TELEMETRY_URL)),
+                    account_id: Some("alice@wonderland".parse()?),
+                    basic_auth: Some(Some(BasicAuth {
+                        web_login: WebLogin::new("mad_hatter")?,
+                        password: SmallStr::from_str("ilovetea"),
+                    })),
+                    public_key: Some(PublicKey::from_str(
+                        "ed01207233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0",
+                    )?),
+                    private_key: Some(PrivateKey::from_hex(
+                        Algorithm::Ed25519,
+                        "9ac47abf59b356e0bd7dcbbbb4dec080e302156a48ca907e47cb6aea1d32719e7233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0"
+                    )?),
+                    ..ConfigurationProxy::default()
+                }
+                .build()?;
+                writeln!(writer, "{}", serde_json::to_string_pretty(&config)?)
+                    .wrap_err("Failed to write.")
```

Comment:
```suggestion
.wrap_err("Failed to write serialized config to the buffer.")
```
And maybe in a few other places throughout the file too.

Reply:
Good suggestion, thanks.